spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -155,6 +155,13 @@
 
 	private static final SpelExpressionParser PARSER = new SpelExpressionParser();
 
+	/*
+	 * Not static as normal since we want this TL to be scoped within the template instance.
```

Comment:
I think your understanding is incorrect, while we normally use static TLs, it's not required if you want a TL per instance...

	public static class Foo {

		static ThreadLocal<String> foo = new ThreadLocal<>();

		ThreadLocal<String> bar = new ThreadLocal<>();

	}

        ...

	Foo one = new Foo();
	Foo.foo.set("foo");
	one.bar.set("bar");
	Foo two = new Foo();
	System.out.println(two.foo.get() + "/" + two.bar.get());

    foo/null

In this case, we need an instance per template because they might invoke a template that's configured to use a different connection factory.